src/sources.cpp

```diff
@@ -478,9 +484,16 @@ void fields::add_volume_source(component c, const src_time &src, const volume &w
   src_vol_chunkloop_data data;
   data.A = A ? A : one;
   data.amp = amp;
+  // correct units for J delta-function amplitude
   LOOP_OVER_DIRECTIONS(gv.dim, d) {
-    if (where.in_direction(d) == 0.0 && !nosize_direction(d)) // delta-fun
-      data.amp *= gv.a; // correct units for J delta-function amplitude
+    if (where.in_direction(d) == 0.0 && !nosize_direction(d)) { // delta-fun
+      if (d == R && where.center().in_direction(d) > 0)
+        // in cylindrical coordinates, we need to scale by $r$ for a dipole.
+        data.amp *= 1.0 / where.center().in_direction(d);
```

Comment:
Why was the scaling by $\Delta r \times \Delta z$ removed?

Reply:
See the code comment below:

```
        // we handle the resolution scaling at the chunk level for cylindrical coordinates.
```